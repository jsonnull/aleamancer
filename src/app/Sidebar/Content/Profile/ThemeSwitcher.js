/* @flow */
import React from 'react'
import Label from '../Label'
import Button from 'app/components/Button'
import styles from './style.css'
import type { Theme } from 'types'

type ButtonProps = {
  selected: boolean,
  onClick: Function,
  children?: any
}
const ThemeButton = (props: ButtonProps) => {
  let variant = (props.selected) ? 'green' : 'outline'
  return <Button
      customClass={styles.themeButton}
      variant={variant}
      onClick={() => props.onClick()}>
    { props.children }
  </Button>
}

type Props = {
  currentTheme: Theme,
  changeTheme: (Theme) => any
}

const ThemeSwitcher = (props: Props) => {
  return <div>
    <Label>Theme</Label>
    <div className={ styles.themeButtons }>
      <ThemeButton
          selected={ props.currentTheme == 'light' }
          onClick={ () => props.changeTheme('light') }>
        Light
      </ThemeButton>
      <ThemeButton
          selected={ props.currentTheme == 'dark' }
          onClick={ () => props.changeTheme('dark') }>
        Dark
      </ThemeButton>
    </div>
  </div>
}

export default ThemeSwitcher
