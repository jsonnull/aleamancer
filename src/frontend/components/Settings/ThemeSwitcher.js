// @flow
import React from 'react'
import styled from 'styled-components'
import Label from '../Label'
import Button from '../Button'
import type { ThemeName } from 'common/types'

const DarkButton = Button.extend`
  margin-left: 5px;
`

const ThemeButtons = styled.div`
  display: flex;
`

type Props = {
  currentTheme: ThemeName,
  changeTheme: ThemeName => any
}

const ThemeSwitcher = (props: Props) => {
  const isLightTheme = props.currentTheme == 'light'
  const isDarkTheme = props.currentTheme == 'dark'

  const useGreen = { green: true }
  const useOutline = { outline: true }

  const lightButton = isLightTheme ? useGreen : useOutline
  const darkButton = isDarkTheme ? useGreen : useOutline

  return (
    <div>
      <Label>Theme</Label>
      <ThemeButtons>
        <Button {...lightButton} onClick={() => props.changeTheme('light')}>
          Light
        </Button>
        <DarkButton {...darkButton} onClick={() => props.changeTheme('dark')}>
          Dark
        </DarkButton>
      </ThemeButtons>
    </div>
  )
}

export default ThemeSwitcher
