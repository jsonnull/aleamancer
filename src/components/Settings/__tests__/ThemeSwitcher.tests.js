// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import ThemeSwitcher from '../ThemeSwitcher.js'

describe('Settings ThemeSwitcher component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<ThemeSwitcher currentTheme="light" changeTheme={() => {}} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
