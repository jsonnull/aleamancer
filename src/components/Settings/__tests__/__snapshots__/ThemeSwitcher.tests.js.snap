// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings ThemeSwitcher component renders correctly 1`] = `
<div>
  <label
    className="sc-bdVaJa eHwgaL"
  >
    Theme
  </label>
  <div
    className="sc-bxivhb iamvSI"
  >
    <button
      className="sc-bwzfXH cYzoFg"
      onClick={[Function]}
    >
      Light
    </button>
    <button
      className="sc-htpNat fzgoAl"
      onClick={[Function]}
    >
      Dark
    </button>
  </div>
</div>
`;
